proposals/000-decorator-api.md

```diff
@@ -0,0 +1,322 @@
+```
+
+### Handling User Clicks on Decorations
+
+To allow user clicks/taps on a decoration, you must first check that the `activate` interaction is available for the decoration style:
+
+```swift
+navigator.supportsDecorationInteraction(.activate, forStyle: Decoration.Style.Highlight)
+```
+
+Then, enable the `activate` interaction when creating each `Decoration` object.
+
+```swift
+Decoration(
+    id: "12",
+    locator: Locator(...),
+    style: Decoration.Style.Highlight(),
+    interactions: [.activate]
+)
+```
+
+Finally, listen to the `DecorationActivated` event which will be emitted when the user clicks the decoration. The API depends on the platform.
+
+```swift
+navigator.observe(DecorationActivated) { decoration in
+    // Present a highlight pop-up for `decoration`...
+}
+```
+
+### Registering Decoration Styles
+
+Reading apps are welcome to register new decoration styles for custom use cases. The API depends on each Navigator, but here's an example using an HTML Navigator (e.g. for EPUB) to implement a sidemark:
+
+```swift
+// Declare a new style with associated configuration values.
+class Decoration.Style.Sidemark {
+    let tint: Color?
+}
+
+// Create a new HTML decoration template which will be used with our new style.
+let sidemarkTemplate = HTMLDecorationTemplate(
+    layout: .bounds,
+    width: .page,
+    element: { decoration in
+        let style = decoration.style as? Decoration.Style.Sidemark
+        let tint = style?.tint ?? Color.red
+        """
+        <div><div class="sidemark" style="--tint: \(tint.cssColor)"/></div>
+        """,
+    },
+    stylesheet: """
+        .sidemark {
+            float: left;
+            width: 5px;
+            height: 100%;
+            background-color: var(--tint);
+            margin-left: 20px;
+            border-radius: 3px;
+        }
+        [dir=rtl] .sidemark {
+            float: right;
+            margin-left: 0px;
+            margin-right: 20px;
+        }
+        """
+)
+
+// Associate the style with our template when creating the HTML navigator.
+var config = HTMLNavigator.Configuration()
+config.decorationStyles[Decoration.Style.Sidemark] = sidemarkTemplate
+let navigator = HTMLNavigator(config)
+```
+
+![](assets/000-sidemark.png)
+
+### Checking Whether a Navigator Supports a Decoration Style
+
+You should check whether the Navigator supports drawing the decoration styles required by a particular feature before enabling it. For example, underlining an audiobook does not make sense, so an Audiobook Navigator would not support the `underline` decoration style.
+
+```swift
+navigator.supportsDecorationStyle(Decoration.Style.Underline)
+```
+
+### Backward Compatibility and Migration
+
+#### Kotlin
+
+The Readium Kotlin toolkit currently ships with a highlighting API. It will be deprecated in favor of the new Decorator API. The old APIs will internally use the Decorator implementation, so it will not be a breaking change.
```

Comment:
That's right, mainly because we need this fast at DM. This would also help get rid of the activities in the test app since we won't need `R2EpubActivity` anymore.